src/refresh_sentinel.rs

```diff
@@ -0,0 +1,311 @@
+//
+// By contributing code to the SAFE Network Software, or to this project generally, you agree to be
+// bound by the terms of the MaidSafe Contributor Agreement, version 1.0.  This, along with the
+// Licenses can be found in the root directory of this project at LICENSE, COPYING and CONTRIBUTOR.
+//
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.
+
+#![forbid(bad_style, missing_docs, warnings)]
+#![deny(deprecated, improper_ctypes, non_shorthand_field_patterns,
+        overflowing_literals, plugin_as_library, private_no_mangle_fns, private_no_mangle_statics,
+        raw_pointer_derive, stable_features, unconditional_recursion, unknown_lints,
+        unsafe_code, unsigned_negation, unused, unused_allocation, unused_attributes,
+        unused_comparisons, unused_features, unused_parens, while_true)]
+#![warn(trivial_casts, trivial_numeric_casts, unused_extern_crates, unused_import_braces,
+        unused_qualifications, variant_size_differences)]
+
+extern crate lru_time_cache;
+use lru_time_cache::LruCache;
+
+/// Entry for accumulation.
+#[derive(Clone)]
+pub struct Entry<V> {
+    /// Values accumulated for a given key.
+    pub received_response: Vec<V>,
+}
+
+/// Generic type for accumulating multiple values under a given key.
+#[allow(dead_code)]
+pub struct RefreshSentinel<K, V> where K: PartialOrd + Ord + Clone, V: Clone {
+    /// Threshold for resolution.
+    quorum: usize,
+    storage: LruCache<K, Entry<V>>
+}
+
+impl<K: PartialOrd + Ord + Clone, V: Clone> RefreshSentinel<K, V> {
+    /// Construct with quorum.
+    #[allow(dead_code)]
+    pub fn new(quorum: usize) -> RefreshSentinel<K, V> {
+        RefreshSentinel { quorum: quorum, storage: LruCache::<K, Entry<V>>::with_capacity(1000) }
+    }
+
+    /// Check for the existence of a key.
+    #[allow(dead_code)]
+    pub fn contains_key(&self, key: &K) -> bool {
+        self.storage.check(key)
+    }
+
+    /// Check whether a quorum of values has been accumulated for the given key.
+    #[allow(dead_code)]
+    pub fn is_quorum_reached(&mut self, key: &K) -> bool {
+        let entry = self.storage.get(key);
+
+        if entry.is_none() {
+            false
+        } else {
+            entry.unwrap().received_response.len() >= self.quorum
+        }
+    }
+
+    /// Adds a key/value pair, if the key already exists add the value under that key.
+    /// Optionally returns the key and the vector of values if the quroum has been reached. 
+    #[allow(dead_code)]
+    pub fn add(&mut self, key: K, value: V)-> Option<(K, Vec<V>)> {
+        let entry = self.storage.remove(&key);
+        if entry.is_none() {
+            let entry_in = Entry { received_response : vec![value]};
+            self.storage.add(key.clone(), entry_in.clone());
```

Comment:
we don't need to clone() entry_in here, same for key - I think, Im not rustc of course - moving the value suffices